extension.js

```diff
@@ -17,584 +17,614 @@
+                        this._updateProcess_stream = new Gio.DataInputStream({
+                            base_stream: new Gio.UnixInputStream({fd: out_fd}),
+                        });
+                        // We will process the output at once when it's done
+                        this._updateProcess_sourceId = GLib.child_watch_add(0, pid, () => {
+                            this._checkUpdatesRead();
+                        });
+                        this._updateProcess_pid = pid;
+                    } catch (err) {
+                        this._showChecking(false);
+                        this.lastUnknowErrorString = err.message.toString();
+                        this._updateStatus(-2);
+                    }
+                }
+
+                _cancelCheck() {
+                    if (this._updateProcess_pid == null)
+                        return;
+
+
+                    Util.spawnCommandLine(`kill ${this._updateProcess_pid}`);
+                    this._updateProcess_pid = null; // Prevent double kill
+                    this._checkUpdatesEnd();
+                }
+
+                _checkUpdatesRead() {
+                    // Read the buffered output
+                    let updateList = [];
+                    let out, size;
+                    do {
+                        [out, size] = this._updateProcess_stream.read_line_utf8(null);
+                        if (out)
+                            updateList.push(out);
+                    } while (out);
+                    this._updateList = updateList;
+                    this._checkUpdatesEnd();
+                }
+
+                _checkUpdatesEnd() {
+                    // Free resources
+                    this._updateProcess_stream.close(null);
+                    this._updateProcess_stream = null;
+                    GLib.source_remove(this._updateProcess_sourceId);
+                    this._updateProcess_sourceId = null;
+                    this._updateProcess_pid = null;
+                    // Update indicator
+                    this._showChecking(false);
+                    if (DISABLE_PARSING) {
+                        this._updateStatus(this._updateList.length);
+                    } else {
+                        this._updateStatus(this._updateList.filter(function (line) {
+                            return RE_UpdateLine.test(line);
+                        }).length);
+                    }
+                }
+
+                _showNotification(title, message) {
+                    if (this._notifSource == null) {
+                        // We have to prepare this only once
+                        this._notifSource = new MessageTray.SystemNotificationSource();
+                        this._notifSource.createIcon = function () {
+                            let gicon = Gio.icon_new_for_string(`${archUpdateIndicatorExtension.dir.get_child('icons').get_path()}/arch-updates-logo.svg`);
```

Comment:
Don't change existing code without good reason, and related to PR
I get the change from Me to archUpdateIndicatorExtension though.